Simplified/Adobe DRM/AdobeDRMContainer.mm

```diff
@@ -0,0 +1,156 @@
+//
+//  AdobeDRMContainer.mm
+//  SimplyE
+//
+//  Created by Vladimir Fedorov on 13.05.2020.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+#import "AdobeDRMContainer.h"
+
+#include <memory>
+#import <ePub3/nav_table.h>
+#import <ePub3/container.h>
+#import <ePub3/initialization.h>
+#import <ePub3/utilities/byte_stream.h>
+#import <ePub3/utilities/error_handler.h>
+#import "adept_filter.h"
+#import "ADEPT/ADEPT.h"
+#import "ePub3/utilities/make_unique.h"
+#import "DataByteStream.h"
+
+static id acsdrm_lock = nil;
+
+@interface AdobeDRMContainer () {
+    @private std::shared_ptr<ePub3::Container> container;
+    @private ePub3::Package *package;
+    @private ePub3::ConstManifestItemPtr manifestItem;
+}
+@end
+
+
+@implementation AdobeDRMContainer: NSObject
+
+- (instancetype)initWithURL:(NSURL *)fileURL {
+        
+    if (self = [super init]) {
+        
+        acsdrm_lock = [[NSObject alloc] init];
+        
+//        NSLog(@"ACSDRMContainer initWithURL: %@", fileURL.path);
+        
+        NSString *path = fileURL.path;
+        
+        ePub3::ErrorHandlerFn sdkErrorHandler = ^(const ePub3::error_details& err) {
+            BOOL isSevereEpubError = NO;
+            if (err.is_spec_error()
+                    && (err.severity() == ePub3::ViolationSeverity::Critical
+                    || err.severity() == ePub3::ViolationSeverity::Major))
+                isSevereEpubError = YES;
+
+            return ePub3::DefaultErrorHandler(err);
+        };
+        ePub3::SetErrorHandler(sdkErrorHandler);
+
+        ePub3::InitializeSdk();
+        ePub3::PopulateFilterManager();
+        ePub3::AdeptFilter::Register();
+
+        try {
+            container = ePub3::Container::OpenContainer(path.UTF8String);
+            
+        }
+        catch (std::exception& e) { // includes ePub3::ContentModuleException
+            auto msg = e.what();
+            std::cout << msg << std::endl;
+        }
+        catch (...) {
+        }
```

Comment:
the catch block above catches everything std::exception, not sure what this part does... need to take a closer look here ). this part of initialisation was taken from R1 init function, tried to avoid changing anything there more than it was needed

Reply:
Understood, feel free to note what you just said here in the file, so that we have that context in the code later on.

Reply:
I decided to log `"Unknown error"` there, as a good indicator that we get to this part of code for some reason

Reply:
removed ePub3 code

Reply:
cool! so we don't depend on ePub3 at all now?

Reply:
yes, works directly with Adobe DRM Connector

Reply:
this is awesome, thank you!